aptos-move/framework/aptos-token/sources/token_event_store.move

```diff
@@ -0,0 +1,340 @@
+        creator_addr: address,
+        collection_name: String,
+        old_maximum: u64,
+        new_maximum: u64,
+    }
+
+    //
+    // Token transfer related events
+    //
+
+    /// Event emitted when an user opt-in the direct transfer
+    struct OptInTransferEvent has drop, store {
+        /// True if the user opt in, false if the user opt-out
+        opt_in: bool
+    }
+
+    //
+    // Token mutation events
+    //
+
+    /// Event emitted when the tokendata uri mutates
+    struct UriMutationEvent has drop, store {
+        creator: address,
+        collection: String,
+        token: String,
+        old_uri: String,
+        new_uri: String,
+    }
+
+    /// Event emitted when mutating the default the token properties stored at tokendata
+    struct DefaultPropertyMutateEvent has drop, store {
+        creator: address,
+        collection: String,
+        token: String,
+        keys: vector<String>,
+        /// we allow upsert so the old values might be none
+        old_values: vector<Option<PropertyValue>>,
+        new_values: vector<PropertyValue>,
+    }
+
+    /// Event emitted when the tokendata description is mutated
+    struct DescriptionMutateEvent has drop, store {
+        creator: address,
+        collection: String,
+        token: String,
+        old_description: String,
+        new_description: String,
+    }
+
+    /// Event emitted when the token royalty is mutated
+    struct RoyaltyMutateEvent has drop, store {
+        creator: address,
+        collection: String,
+        token: String,
+        old_royalty_numerator: u64,
+        old_royalty_denominator: u64,
+        old_royalty_payee_addr: address,
+        new_royalty_numerator: u64,
+        new_royalty_denominator: u64,
+        new_royalty_payee_addr: address,
+    }
+
+    /// Event emitted when the token maximum is mutated
+    struct MaxiumMutateEvent has drop, store {
+        creator: address,
+        collection: String,
+        token: String,
+        old_maximum: u64,
+        new_maximum: u64,
+    }
+
+    struct TokenEventStoreV1 has key {
+        /// collection mutation events
+        collection_uri_mutate_events: EventHandle<CollectionUriMutateEvent>,
+        collection_maximum_mutate_events: EventHandle<CollectionMaxiumMutateEvent>,
+        collection_description_mutate_events: EventHandle<CollectionDescriptionMutateEvent>,
+        /// token transfer opt-in event
+        opt_in_events: EventHandle<OptInTransferEvent>,
+        /// token mutation events
+        uri_mutate_events: EventHandle<UriMutationEvent>,
+        default_property_mutate_events: EventHandle<DefaultPropertyMutateEvent>,
+        description_mutate_events: EventHandle<DescriptionMutateEvent>,
+        royalty_mutate_events: EventHandle<RoyaltyMutateEvent>,
+        maximum_mutate_events: EventHandle<MaxiumMutateEvent>,
+        /// This is for adding new events in future
+        extention: Option<Any>,
```

Comment:
extension

Reply:
We probably need to start running a spell check on PRs... Let's get this fixed asap